### PR TITLE
Buy album

### PR DESCRIPTION
**Pull request by yxd99**
## Summary

- Register an album in my account by entering the id in the request.
- The request is rejected if the album was previously purchased (it is already in my account)

## Trello Card

[https://trello.com/c/zY0gdVYV/31-comprar-album](url)

### CODE REVIEW

File: app/models/user.js [hunk 1]
```diff
@@ -34,8 +34,10 @@ module.exports = (sequelize, DataTypes) => {
...
     {}
   );
-  // User.associate = models => {
-  // associations can be defined here
-  // };
+  /*
+  User.associate = models => {
+    // associations can be defined here
+  };
+  */
```
Comment (SOY4RIAS) [inline app/models/user.js]: if you dont need it remove it

File: app/models/sale.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+        type: DataTypes.STRING
+      }
+    },
+    {}
+  );
+  Sale.associate = models => {
+    Sale.belongsTo(models.User, {
+      foreignKey: 'idUser'
+    });
+    // associations can be defined here
```
Comment (SOY4RIAS) [inline app/models/sale.js]: remove

File: app/services/album.js [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+      return album;
+    } catch (err) {
+      throw errors.databaseError(err);
+    }
+  }
+
+  static async getMyAlbumForId(idUser, idAlbum) {
+    try {
+      const album = await db.Sale.findOne({ where: { idUser, idAlbum } });
+      return album || { errorAlbum: 'You have not bought the album.' };
```
Comment (SOY4RIAS) [inline app/services/album.js]:
```suggestion
      return album || { error: 'You have not bought the album.' };
```
think over the consistency related to your other methods and it is redundant to call it "errorAlbum"

File: app/constants.js [hunk 4]
```diff
@@ -5,4 +5,8 @@ exports.ROLES = {
 
 exports.AUTH_HEADER = 'Authorization';
 
-exports.ALBUM_SUPPLIER = 'https://jsonplaceholder.typicode.com/albums';
+exports.ALBUM_SUPPLIER = 'https://jsonplaceholder.typicode.com/';
+
+exports.OPT_SUPPLIER_JP = {
```
Comment (SOY4RIAS) [inline app/constants.js]: what means OPT_SUPPLIER_JP?

**Review by SOY4RIAS — CHANGES_REQUESTED** [review]

File: app/constants.js (hunk 4 above)
Comment (yxd99) [reply app/constants.js]: OPTIONS SUPPLIER JSONplaceholder

**Pull request closed without merging**